ts/src/time-profiler.ts

```diff
@@ -49,40 +49,36 @@ export interface TimeProfilerOptions {
 }
 
 export async function profile(options: TimeProfilerOptions) {
-  const stop = start(
+  const stop = await start(
     options.intervalMicros || DEFAULT_INTERVAL_MICROS,
-    options.name,
-    options.sourceMapper,
-    options.lineNumbers
+    options.sourceMapper
   );
   await delay(options.durationMillis);
-  return stop();
+  return await stop();
 }
 
-export function start(
+export async function start(
```

Comment:
This now has to be async which is another breaking change.

Reply:
There's a chance (depending on underlying implementation) this will result in a fix for https://github.com/googleapis/cloud-profiler-nodejs/issues/683 (event loop blocked at start of profile collection). 

This would be a good change if it does that (though needing to update agent code will be a bummer)

Reply:
Great! I will assume this fixes that issue and close it once this change gets pull into that repo.